alma/lib/API/RefundHelper.php

```diff
@@ -0,0 +1,102 @@
+<?php
+/**
+ * 2018-2023 Alma SAS
+ *
+ * THE MIT LICENSE
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+ * Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+ * WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ *
+ * @author    Alma SAS <contact@getalma.eu>
+ * @copyright 2018-2023 Alma SAS
+ * @license   https://opensource.org/licenses/MIT The MIT License
+ */
+
+namespace Alma\PrestaShop\API;
+
+use Alma\PrestaShop\Utils\Logger;
+use Cart;
+
+if (!defined('_PS_VERSION_')) {
+    exit;
+}
+
+/**
+ * Class RefundHelper.
+ *
+ * Use for refund
+ */
+class RefundHelper
+{
+    /** @var Alma */
+    private $module;
+    /** @var Cart */
+    private $cart;
+    /** @var ClientHelper */
+    private $almaClient;
+    /** @var int */
+    private $paymentId;
+
+    /**
+     * construct RefundHelper
+     *
+     * @param Alma $module
+     * @param Cart $cart
+     * @param int $paymentId
+     */
+    public function __construct($module, $cart, $paymentId)
+    {
+        $this->module = $module;
+        $this->cart = $cart;
+        $this->almaClient = new ClientHelper();
+        $this->paymentId = $paymentId;
+    }
+
+    /**
+     * Refund for Mismatch
+     *
+     * @return void
+     *
+     * @throws RefundException
+     * @throws PaymentValidationError
+     */
+    public function forMismatch()
```

Comment:
I think the function name isn't the best: something like  $refundHelper->mismatchAmountRefund();?

Reply:
I thought not to put `Refund` in the name function because we are in the class refundHelper. We see the refund in $refundHelper->
WDYT ?